modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/internal/graphql/contributor/test/BaseGraphQLContributor.java

```diff
@@ -0,0 +1,208 @@
+				this.id = id;
+			}
+
+			@com.liferay.portal.vulcan.graphql.annotation.GraphQLField
+			protected long id;
+
+		}
+
+		@GraphQLTypeExtension(MockDTO.class)
+		public class MockGraphQLTypeExtension {
+
+			public MockGraphQLTypeExtension(MockDTO mockDTO) {
+				_mockDTO = mockDTO;
+			}
+
+			@com.liferay.portal.vulcan.graphql.annotation.GraphQLField
+			public String field() {
+				return _testString;
+			}
+
+			private final MockDTO _mockDTO;
+
+		}
+
+		private static long _testId;
+		private static String _testString;
+
+	}
+
+	public static class MockServletDataImpl implements ServletData {
+
+		public MockServletDataImpl(long testId, String testString) {
+			MockQuery.setTestId(testId);
+			MockQuery.setTestString(testString);
+		}
+
+		@Override
+		public Object getMutation() {
+			return null;
+		}
+
+		@Override
+		public String getPath() {
+			return "/mock-path-graphql/v1.0";
+		}
+
+		@Override
+		public Object getQuery() {
+			return new MockQuery();
+		}
+
+	}
+
+	protected JSONObject invokeGraphQLQuery(GraphQLField graphQLField)
+		throws Exception {
+
+		GraphQLField queryGraphQLField = new GraphQLField(
+			"query", graphQLField);
+
+		return _invoke(queryGraphQLField.toString(), Http.Method.POST);
+	}
+
+	protected static class GraphQLField {
+
+		public GraphQLField(String key, GraphQLField... graphQLFields) {
+			this(key, new HashMap<>(), graphQLFields);
+		}
+
+		public GraphQLField(
+			String key, Map<String, Object> parameterMap,
+			GraphQLField... graphQLFields) {
+
+			_key = key;
+			_parameterMap = parameterMap;
+			_graphQLFields = Arrays.asList(graphQLFields);
+		}
+
+		@Override
+		public String toString() {
+			StringBuilder sb = new StringBuilder(_key);
+
+			if (!_parameterMap.isEmpty()) {
+				sb.append("(");
+
+				for (Map.Entry<String, Object> entry :
+						_parameterMap.entrySet()) {
+
+					sb.append(entry.getKey());
+					sb.append(": ");
+					sb.append(entry.getValue());
+					sb.append(", ");
+				}
+
+				sb.setLength(sb.length() - 2);
+
+				sb.append(")");
+			}
+
+			if (!_graphQLFields.isEmpty()) {
+				sb.append("{");
+
+				for (GraphQLField graphQLField : _graphQLFields) {
+					sb.append(graphQLField.toString());
+					sb.append(", ");
+				}
+
+				sb.setLength(sb.length() - 2);
+
+				sb.append("}");
+			}
+
+			return sb.toString();
+		}
+
+		private final List<GraphQLField> _graphQLFields;
+		private final String _key;
+		private final Map<String, Object> _parameterMap;
+
+	}
+
+	private JSONObject _invoke(String query, Http.Method httpMethod)
+		throws Exception {
+
+		Http.Options options = new Http.Options();
+
+		options.addHeader(
+			HttpHeaders.CONTENT_TYPE, ContentTypes.APPLICATION_JSON);
+		options.addHeader(
+			"Authorization",
+			"Basic " + Base64.encode("test@liferay.com:test".getBytes()));
+		options.setLocation("http://localhost:8080/o/graphql");
+		options.setMethod(httpMethod);
+		options.setBody(
+			new Http.Body(
+				JSONUtil.put(
+					"query", query
```

Comment:
maybe it should be useful to make configurable the type of parameter, but it could be done in the future if someone needs to test a mutation instead